Source/JavaScriptCore/wasm/js/JSWebAssemblyStruct.h

```diff
@@ -31,22 +31,24 @@
 #include "WasmTypeDefinitionInlines.h"
 #include "WebAssemblyGCObjectBase.h"
 #include <wtf/Ref.h>
+#include <wtf/TrailingArray.h>
 
 namespace JSC {
 
 class JSWebAssemblyInstance;
 
-class JSWebAssemblyStruct final : public WebAssemblyGCObjectBase {
+class JSWebAssemblyStruct final : public WebAssemblyGCObjectBase, public TrailingArray<JSWebAssemblyStruct, uint8_t> {
```

Comment:
How about inheriting `TrailingArray` via `private` to hide random methods of that from the public?

Reply:
Done, it's a bit awkward you have to `friend TrailingArrayType` in general though...